crates/ruma-client-api/src/discovery/discover_homeserver.rs

```diff
@@ -112,3 +131,163 @@ impl TileServerInfo {
         Self { map_style_url }
     }
 }
+
+/// Information about a specific MatrixRTC focus.
+#[cfg(feature = "unstable-msc4143")]
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[cfg_attr(not(ruma_unstable_exhaustive_types), non_exhaustive)]
+#[serde(tag = "type")]
+pub enum RtcFocusInfo {
+    /// A LiveKit RTC focus.
+    #[serde(rename = "livekit")]
+    LiveKit(LiveKitRtcFocusInfo),
+
+    /// A custom RTC focus.
+    #[doc(hidden)]
+    #[serde(untagged)]
+    _Custom(CustomRtcFocusInfo),
+}
+
+#[cfg(feature = "unstable-msc4143")]
+impl RtcFocusInfo {
+    /// A constructor to create a LiveKit RTC focus.
+    pub fn live_kit(service_url: String) -> Self {
+        Self::LiveKit(LiveKitRtcFocusInfo { service_url })
+    }
+
+    /// A constructor to create a custom RTC focus.
+    pub fn custom(focus_type: String, data: JsonObject) -> Self {
```

Comment:
This should actually be named `new`, and construct the proper variant in case `focus_type` is `livekit`. See also the implementation of `MessageType::new()`.

Reply:
Ah I see, thanks 🙏